feat: Engagement Center Add permalink to the engagement center tabs - MEEDS-449 - meeds-io/meeds#13

ArfaouiMohamedAmine opened the pull request:
This change is going to add `permalinks` to the `engagement center tabs` in order to be accessed easily using the followings;
`{DOMAIN}/portal/{SITE}/engagement/challenges`
`{DOMAIN}/portal/{SITE}/engagement/programs`
`{DOMAIN}/portal/{SITE}/engagement/achievement`

ArfaouiMohamedAmine committed:
MEED-449 : adding permalink to the engagement center tabs (#428)

This change is going to add `permalinks` to the `engagement center tabs` in order to be accessed easily using the followings;
{DOMAIN}/portal/{SITE}/engagement/challenges
{DOMAIN}/portal/{SITE}/engagement/programs
{DOMAIN}/portal/{SITE}/engagement/achievement

plamarque commented on challenges-webapp/src/main/resources/locale/navigation/portal/global_en.properties:
```diff
@@ -15,3 +15,5 @@
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
 portal.global.challenges=Challenges
+portal.global.engagement=Engagement
```
```suggestion
portal.global.contributions=Contributions
```

plamarque commented on challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/app-center-configuration.xml:
```diff
@@ -59,7 +59,7 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
                         </field>
                         <field name="url">
-                            <string>./challenges</string>
+                            <string>./engagement</string>
```
```suggestion
                            <string>./contributions</string>
```

plamarque commented on challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml:
```diff
@@ -30,5 +30,11 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
             <page-reference>portal::global::challenges</page-reference>
         </node>
+        <node>
+            <name>engagement</name>
```
```suggestion
            <name>contributions</name>
```

plamarque commented on challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml:
```diff
@@ -30,5 +30,11 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         </node>
+        <node>
+            <name>engagement</name>
+            <label>#{portal.global.engagement}</label>
```
```suggestion
            <label>#{portal.global.contributions}</label>
```

plamarque commented on challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml:
```diff
@@ -30,5 +30,11 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+            <name>engagement</name>
+            <label>#{portal.global.engagement}</label>
+            <visibility>SYSTEM</visibility>
+            <page-reference>portal::global::engagement</page-reference>
```
```suggestion
            <page-reference>portal::global::contributions</page-reference>
```

plamarque commented on challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/pages.xml:
```diff
@@ -21,8 +21,8 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
   <page>
-    <name>challenges</name>
-    <title>challenges</title>
+    <name>engagement</name>
```
```suggestion
    <name>contributions</name>
```

plamarque commented on challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/pages.xml:
```diff
@@ -21,8 +21,8 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-    <name>challenges</name>
-    <title>challenges</title>
+    <name>engagement</name>
+    <title>engagement</title>
```
```suggestion
    <title>contributions</title>
```

plamarque commented on challenges-webapp/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeDetailsDrawer.vue:
```diff
@@ -179,7 +179,7 @@ export default {
     open(challenge) {
       this.challenge = challenge;
-      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/challenges/${this.challenge.id}`);
+      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/engagement/challenges/${this.challenge.id}`);
```
```suggestion
      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/challenges/${this.challenge.id}`);
```

plamarque commented on challenges-webapp/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeDetailsDrawer.vue:
```diff
@@ -195,7 +195,7 @@ export default {
     },
     close() {
-      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/challenges`);
+      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/engagement/challenges`);
```
```suggestion
      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/challenges`);
```

plamarque reviewed (commented):
aSuggestion to adopt the term "Contributions" (Contributions Center, contribution program, etc.) instead of Engagement.

CC @Samuel-Renault-eXo

srenault-meeds commented:
> 

Let's stick to Engagement for now and we will propose one task to review the labelling once it is validated.

AzmiTouil commented:
> > 
> 
> Let's stick to Engagement for now and we will propose one task to review the labelling once it is validated.

I think we need to work on it in the same task, because it's not just the labeling, it's also changing the page name and the navigation name

ArfaouiMohamedAmine committed:
feat: Engagement Center Add permalink to the engagement center tabs - MEEDS-449 - meeds-io/meeds#13 (#466)

This change is going to add `permalinks` to the `engagement center` tabs
in order to be accessed easily using the followings;
`{DOMAIN}/portal/{SITE}/engagement/challenges`
`{DOMAIN}/portal/{SITE}/engagement/programs`
`{DOMAIN}/portal/{SITE}/engagement/achievement`

ArfaouiMohamedAmine commented:
@AzmiTouil ready for review

AzmiTouil committed:
Merge branch 'develop-meed' into backport/MEED-449
